### PR TITLE
Add RGBD + reduce changes between hkr and development

### PR DESCRIPTION
**Pull request by SamerKhshiboun**
- add RGBD to development
- Reduce changes between hkr and development

### CODE REVIEW

File: realsense2_camera/src/rs_node_setup.cpp [hunk 1]
```diff
@@ -303,6 +307,27 @@ void BaseRealSenseNode::startPublishers(const std::vector<stream_profile>& profi
...
+{
+    _rgbd_publisher.reset();
+    if(_enable_rgbd && !_rgbd_publisher)
+    {
+        if (_sync_frames && _is_color_enabled && _is_depth_enabled)
+        {
+            rmw_qos_profile_t qos = rmw_qos_profile_system_default;
+
+            rclcpp::PublisherOptionsWithAllocator<std::allocator<void>> options;
+            options.use_intra_process_comm = rclcpp::IntraProcessSetting::Disable;
```
Comment (Nir-Az) [inline realsense2_camera/src/rs_node_setup.cpp]: Check how can we support `enable`?
On depth we support it

**Comment by Nir-Az** [comment]
Can we add some tests for RGBD?

File: realsense2_camera/src/rs_node_setup.cpp (hunk 1 above)
Comment (SamerKhshiboun) [reply realsense2_camera/src/rs_node_setup.cpp]: Fixed, and tested locally.

**Comment by SamerKhshiboun** [comment]
> Can we add some tests for RGBD?

I added a separate ticket for this.
I want to align between the two branches.
Then I will start adding tests.

**Review by Nir-Az — APPROVED** [review]

**Pull request merged**